casper/src/main/scala/coop/rchain/casper/genesis/contracts/ProofOfStake.scala

```diff
@@ -32,16 +34,31 @@ final case class ProofOfStake(minimumBond: Long, maximumBond: Long, validators:
 
   val path: String = "<synthetic in ProofOfStake.scala>"
 
-  val code: String =
-    s"""
-       | new rl(`rho:registry:lookup`), PoSCh in {
-       |   rl!(`rho:id:cnec3pa8prp4out3yc8facon6grm3xbsotpd4ckjfx8ghuw77xadzt`, *PoSCh)
-       |   | for (@(_, PoS) <- PoSCh) {
-       |     @PoS!($minimumBond, $maximumBond)
-       |   }
-       | }
-     """.stripMargin
+  val code: String = loadTemplateFromResources(
+    "PoS.rhox",
+    "minimumBond" -> minimumBond,
+    "maximumBond" -> maximumBond
+  )
 
   lazy val term: Par = ParBuilder[Coeval].buildNormalizedTerm(code).value()
 
+  /**
+    * Loads code from a resource file while doing macro substitution with the provided values.
+    * The macros have the format $$macro$$
+    * @param file
+    * @param env a sequence of pairs macro -> value
+    * @return
+    */
+  private def loadTemplateFromResources(file: String, env: (String, Any)*): String = {
+    val originalContent = Source.fromResource(file).mkString
+
+    val finalContent = env.foldLeft(originalContent) {
+      case (content, (name, value)) => content.replace(s"""$$$$$name$$$$""", value.toString)
+    }
+
+    s"""//Loaded from resource file <<$file>>
+       |$finalContent
+       |""".stripMargin
```

Comment:
this should probably be

```scala
    s"""//Loaded from resource file <<$file>>
       #$finalContent
       #""".stripMargin('#')
```

or else we have veeeeery weird errors as soon as s/b leaves a `|` as a first character in a line (no matter how deeply indented). We've had quite some debugging with @JosDenmark around exactly this.